packages/components/dashboard/inventory-card/inventory-card.tsx

```diff
@@ -0,0 +1,388 @@
+      errorList,
+      processingList,
+    };
+  }, [buckets, bucketsLoaded, bucketsError]);
+
+  const {
+    statusMap: dataResourceStatusMap,
+    errorList: dataResourceErrorList,
+    processingList: dataResourceProcessingList,
+  } = React.useMemo((): displayItems => {
+    const errorList = [];
+    const processingList = [];
+    const statusMap =
+      dataResourcesLoaded && !dataResourcesError
+        ? dataResources?.reduce((drMap, dr) => {
+            const dataResourceStatus =
+              NamespaceStorePhaseMap[dr?.status?.phase];
+            const dataResourceName = dr?.metadata?.name;
+            if (dataResourceStatus === PhaseType.ERROR) {
+              errorList.push(dataResourceName);
+            } else if (dataResourceStatus === PhaseType.PROCESSING) {
+              processingList.push(dataResourceName);
+            }
+            drMap[dataResourceStatus] = drMap[dataResourceStatus] + 1 || 1;
+            return drMap;
+          }, {})
+        : {};
+    return {
+      statusMap,
+      errorList,
+      processingList,
+    };
+  }, [dataResources, dataResourcesLoaded, dataResourcesError]);
+
+  const {
+    statusMap: obClaimsStatusMap,
+    errorList: obcErrorList,
+    processingList: obcProcessingList,
+  } = React.useMemo((): displayItems => {
+    const errorList = [];
+    const processingList = [];
+    const statusMap =
+      obcLoaded && !obcError
+        ? obc?.reduce((obcMap, obc) => {
+            const obClaimsStatus =
+              ObjectBucketClaimPhaseMap[obc?.status?.phase];
+            const obcResource: obcType = {
+              name: obc?.metadata?.name,
+              ns: obc?.metadata?.namespace,
+            };
+            if (obClaimsStatus === PhaseType.ERROR) {
+              errorList.push(obcResource);
+            } else if (obClaimsStatus === PhaseType.PROCESSING) {
+              processingList.push(obcResource);
+            }
+            obcMap[obClaimsStatus] = obcMap[obClaimsStatus] + 1 || 1;
+            return obcMap;
+          }, {} as { [key: string]: number })
+        : {};
+
+    return {
+      statusMap,
+      errorList,
+      processingList,
+    };
+  }, [obc, obcLoaded, obcError]);
+
+  return (
+    <Card>
+      <CardHeader>
+        <CardTitle>{t('Inventory')}</CardTitle>
+      </CardHeader>
+      <CardBody>
+        <HealthBody>
+          <Gallery
+            className="co-overview-status__health inventory-card-body"
+            hasGutter
+          >
+            <CustomGalleryItem
+              listItems={buckets}
+              listLoaded={bucketsLoaded}
+              listError={bucketsError}
+              redirectPath={BUCKET_CLASS_LIST_PATH}
+              resource={t('Buckets')}
+              statusMap={bucketStatusMap}
+              errorPopup={
+                <BucketPopup
+                  type={PhaseType.ERROR}
+                  statusmap={bucketStatusMap}
+                  data={bucketClassErrorList}
+                />
```

Comment:
```suggestion
```
don't pass entire component as a prop. Pass objects/list/variables... and call these pop component directly in `CustomGalleryItem`